src/progress_bar.rs

```diff
@@ -948,8 +803,9 @@ mod tests {
 
         let state = mp.state.read().unwrap();
         // the removed place for p1 is reused
-        assert_eq!(state.objects.len(), 2);
-        assert_eq!(state.objects.iter().filter(|obj| obj.is_some()).count(), 1);
+        assert_eq!(state.draw_states.len(), 2);
+        assert_eq!(state.free_set.len(), 1);
+        assert_eq!(state.draw_states.len() - state.free_set.len(), 1);
```

Comment:
Let's add a `len()` method to `MultiProgressState` to encapsulate this logic?

Reply:
Done, I added some asserts that in `push` and `remove` to check that the two lengths match, mostly to have a non test use for the new `len` method to keep the dead code check happy, but also as I think it is a good sanity check.

Reply:
Nice, thanks.